File: Week 2/Problem Set 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(const char *str);
int count_sentences(const char *str);
int count_words(const char *str);

int main(void)
{
    char text[10000];
    printf("Text: ");
    fgets(text, sizeof(text), stdin);

    double W = count_words(text);
    double L = count_letters(text) / W * 100;
    double S = count_sentences(text) / W * 100;

    double index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

    // printf("letters %f\n", L);
    // printf("words %f\n", W);
    // printf("sentences %f\n", S);
}

int count_letters(const char *str)
{
    int count = 0;

    while (*str != '\0')
    {
        if (isalpha(*str))
            count++;
        str++;
    }
    return count;
}

int count_sentences(const char *str)
{
    int count = 0;

    while (*str != '\0')
    {
        if (*str == '.' || *str == '!' || *str == '?')
            count++;
        str++;
    }
    return count;
}

int count_words(const char *str)
{
    int count = 0;
    if (*str != '\0')
        count++;
    for (int i = 0; str[i] != '\0'; i++)
    {
        if (str[i] == ' ' && str[i + 1] != ' ')
        {
            count++;
            continue;
        }
    }
    return count;
}

File: Week 6/sentimental-readability/readability.py
import sys


def get_input():
    s = input("Text: ")
    return s


def count_letters(text):
    text = str(text)
    L = 0
    for letter in text:
        if letter.isalpha():
            L += 1
    return L


def count_sentences(text):
    S = 0
    text = str(text)
    for char in text:
        if char in [".", "!", "?"]:
            S += 1
    return S


def count_words(text):
    text = str(text)
    return len(text.split())


def main():
    text = get_input()
    W = count_words(text)
    L = count_letters(text) / W * 100
    S = count_sentences(text) / W * 100

    grade = 0.0588 * L - 0.296 * S - 15.8
    grade = round(grade)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade: {grade}")
    return


if __name__ == "__main__":
    main()
